crates/astria-sequencer/src/sequencer.rs

```diff
@@ -210,6 +188,83 @@ fn start_grpc_server(
     )
 }
 
+fn start_abci_server(
+    storage: &cnidarium::Storage,
+    app: App,
+    mempool_service: service::Mempool,
+    listen_url: &str,
+    server_exit_tx: oneshot::Sender<()>,
+) -> Result<JoinHandle<()>, Report> {
+    // Setup services required for the ABCI server
+    let consensus_service = tower::ServiceBuilder::new()
+        .layer(request_span::layer(|req: &ConsensusRequest| {
+            req.create_span()
+        }))
+        .service(tower_actor::Actor::new(10, |queue: _| {
+            let storage = storage.clone();
+            async move { service::Consensus::new(storage, app, queue).run().await }
+        }));
+    let info_service =
+        service::Info::new(storage.clone()).wrap_err("failed initializing info service")?;
+    let snapshot_service = service::Snapshot;
+
+    // Builds the server but does not start listening.
+    let server = Server::builder()
+        .consensus(consensus_service)
+        .info(info_service)
+        .mempool(mempool_service)
+        .snapshot(snapshot_service)
+        .finish()
+        .ok_or_eyre("server builder didn't return server; are all fields set?")?;
+
+    // Validate and parse the listen_url received from the config.
```

Comment:
This (as well as the subsequent server start) could be done more elegantly by providing a type `AbciListenUrl` with variants `Tcp` and `Unix` and implement `FromStr` for it. Then we could also have a few simple unit tests.